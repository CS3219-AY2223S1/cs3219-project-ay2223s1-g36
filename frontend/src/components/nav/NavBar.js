import { List } from '@mui/material';
import { Box } from '@mui/system';
import { matchPath, useLocation } from 'react-router-dom';
import NavItem from './NavItem';

export default function NavBar({ navConfig, ...others }) {
  const { pathname } = useLocation();
  const isActivePage = (path) => (path ? !!matchPath({ path, end: false }, pathname) : false);

  return (
    <Box {...others}>
      <List disablePadding sx={{ p: 1 }}>
        {navConfig.map((item) => (
          <NavItem key={item.title} item={item} active={isActivePage} />
        ))}
      </List>
    </Box>
  );
}
