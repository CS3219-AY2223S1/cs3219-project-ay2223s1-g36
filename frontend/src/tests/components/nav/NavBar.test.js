import renderer from 'react-test-renderer';
import NavBar from '../../../components/nav/NavBar';
import { BrowserRouter as Router } from 'react-router-dom';
import NAV_CONFIG from '../../../components/nav/NavConfig';

it('NavBar renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <NavBar navConfig={NAV_CONFIG} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
