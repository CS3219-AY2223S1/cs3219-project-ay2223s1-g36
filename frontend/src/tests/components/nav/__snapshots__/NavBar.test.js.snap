// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  className="MuiBox-root css-0"
>
  <ul
    className="MuiList-root css-1lu02n7-MuiList-root"
  >
    <a
      className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-80lifd-MuiButtonBase-root-MuiListItemButton-root"
      components={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      href="/dashboard"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      to="/dashboard"
    >
      <div
        className="MuiListItemIcon-root css-xh62yl-MuiListItemIcon-root"
      >
        <span />
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        dashboard
      </div>
    </a>
    <a
      className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-80lifd-MuiButtonBase-root-MuiListItemButton-root"
      components={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      href="/history"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      to="/history"
    >
      <div
        className="MuiListItemIcon-root css-xh62yl-MuiListItemIcon-root"
      >
        <span />
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        history
      </div>
    </a>
    <a
      className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-80lifd-MuiButtonBase-root-MuiListItemButton-root"
      components={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      href="/profile"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      to="/profile"
    >
      <div
        className="MuiListItemIcon-root css-xh62yl-MuiListItemIcon-root"
      >
        <span />
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        profile
      </div>
    </a>
  </ul>
</div>
`;
